apps/src/lab2/progress/ProgressManager.ts

```diff
@@ -6,6 +6,11 @@ import {Condition, Validation} from '@cdo/apps/lab2/types';
 // Abstract class that validates a set of conditions. How
 // the validation works is up to the implementor.
 export abstract class Validator {
+  abstract shouldValidateWithExemplar(): boolean;
+  abstract getExemplarValidationResults(): {
```

Comment:
Optional: can make this non-abstract with a default `false / no message` return value so you don't need to override everywhere, like in the Python validator.

Reply:
Updated!